src/snyk/common/services/learnService.ts

```diff
@@ -0,0 +1,113 @@
+import axios from 'axios';
+import { isCodeIssue, isOssIssue, OpenCommandIssueType } from '../../common/commands/types';
+import type { completeFileSuggestionType } from '../../snykCode/interfaces';
+import { OssIssueCommandArg } from '../../snykOss/views/ossVulnerabilityTreeProvider';
+import { ILog } from '../logger/interfaces';
+
+const ecosystems = [
+  'all',
+  'java',
+  'javascript',
+  'python',
+  'golang',
+  'php',
+  'cpp',
+  'csharp',
+  'kubernetes',
+  'ruby',
+  'elixir',
+  'docker',
+  'npm',
+] as const;
+type LessonEcosystem = typeof ecosystems[number];
+
+const isValidEcosystem = (potentialEcosystem: string | null | undefined): potentialEcosystem is LessonEcosystem => {
+  return ecosystems.includes(potentialEcosystem as LessonEcosystem);
+};
```

Comment:
sounds to me like a function that return boolean

Reply:
It does, this is a 'type predicate'. If you use isValidEcosystem in an if block. Typecript knows it is of type Ecosystem:
https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates